ui/selectors/transactions.js

```diff
@@ -53,6 +54,22 @@ export const getTransactions = createDeepEqualSelector(
   (transactions) => transactions,
 );
 
+export const getAllNetworkTransactions = createDeepEqualSelector(
+  (state) => {
```

Comment:
I feel like we are abusing deep equal here.

IIRC Reselect works by:
1. taking and running the input selectors
2. checks the results of the input selectors
    - if they are the same it returns the memoized result.
    - if they are not the same, then it will compute the output selector.
  
So currently the input selector is the `(state) => {/* process stuff */}`, which is ran on every state change.

Can we refactor this to:
```ts
export const getAllNetworkTransactions = createDeepEqualSelector(
  // Input Selector
  getTransactions,
  // Output selector
  (transactions) => {
    if (!transactions.length) { return [] }
    
    const popularNetworks = FEATURED_NETWORK_CHAIN_IDS
    const filteredTransactions = transactions.filter(...)
  }
)
```